app/models/spree/payment_method/stripe_credit_card.rb

```diff
@@ -61,16 +61,26 @@ def payment_profiles_supported?
         true
       end
 
+      def non_fractional_money(money, currency)
+        amount = money if money.is_a?(Integer)
+        amount ||= money.cents if money.respond_to?(:cents)
+        Spree::Money.new((amount * 100), { currency: currency })
```

Comment:
what would happen if money is neither an Integer nor it responds to :cents?
if that case can never happen, how about using a terniary operator? :)

Reply:
Good point! [It shouldn't ever happen](https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/core/app/models/spree/payment.rb#L82). Thanks. Could be made into a one liner, but two lines seems more clear to me.

Also, in a rush I removed this fix from the 'credit' method after CI failed (I forgot to delete the underscore in the formerly unused param), but it needs to be fixed too because the 'refund' method (calculated [here](https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/core/app/models/spree/payment.rb#L103) in Solidus) still goes through [here] in ActiveMerchant (https://github.com/activemerchant/active_merchant/blob/7fea8f3364ef44187aafa9c5eac3b76438671327/lib/active_merchant/billing/gateway.rb#L279).